worlds/sc2/MissionTables.py

```diff
@@ -18,6 +18,28 @@ class MissionPools(IntEnum):
     FINAL = 5
 
 
+class MissionFlags(IntFlag):
+    none        = 0
+    Terran      = 0x1
+    Zerg        = 0x2
+    Protoss     = 0x4
+    NoBuild     = 0x8
+    Defense     = 0x10
+    Timed       = 0x20   # The overall pace of the mission is guided by timers until victory or defeat
+    Kerrigan    = 0x40   # The player controls Kerrigan in the mission
+    VanillaSoa  = 0x80   # The player controls the Spear of Adun in the vanilla version of the mission
+    Nova        = 0x100  # The player controls NCO Nova in the mission
+    AiAlly      = 0x200  # The mission has an AI ally that can be taken over
```

Comment:
I think this will be insufficient, since AI Ally settings need to know races of allies to include their items. If the ambition is to largely unify all special option cases, the flags probably should be extremely detailed.

As an implementation detail, is there a reason to declare the flag values? [Python docs](https://docs.python.org/3/library/enum.html#enum.IntFlag) say the following:

> Using `auto` with `IntFlag` results in integers that are powers of two, starting with 1.

This would be less pretty, but also less prone to human error I think.

Reply:
I thought about specifying ally race but figured for the first pass it might not be necessary as it's rare enough to special-case the logic wherever it's used. If you're also saying this at first pass, then perhaps it's better to specify it up-front.

In this case I suppose it's not necessary to specify the values. I've just had to do it for interfacing-with-other-files reasons the last few times I've used `IntFlag`s and it became a habit.